### PR TITLE
Brief note about timeouts in the haproxy router

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
Some of the timeouts that we allow cluster administrators to set
interact with others that we expose or in some cases don't expose.
Causing haproxy to actually timeout as the sum of those timeouts

Ex.

setting
ROUTER_SLOWLORIS_HTTP_KEEPALIVE to 10s

because we set "tcp-request inspect-delay" to 5s

ROUTER_SLOWLORIS_HTTP_KEEPALIVE triggers and then the inspect-delay
waits to timeout so the connection closes after 15s

bug 1562244
https://bugzilla.redhat.com/show_bug.cgi?id=1562244

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@bfallonf PTAL

not sure if I should provide a specific example or not

**Comment by akostadinov** [comment]
It's good to have this specific example because it is easy to hit. Also (even more useful) to link to the relevant HAProxy documentation where more detailed information can be found.

File: architecture/topics/router_environment_variables.adoc [hunk 1]
```diff
@@ -81,6 +81,11 @@ endif::[]
 |`ROUTER_CIPHERS` | intermediate  | Specify the set of xref:ciphers[ciphers] supported by bind.
 |===
 
+[NOTE]
+====
+Some of the timeouts that we expose interact casuing the total timeout to be longer then specified.
```
Comment (knobunc) [inline architecture/topics/router_environment_variables.adoc]: Should we say:

e.g. ROUTER_SLOWLORIS_HTTP_KEEPALIVE  adjusts "timeout http-keep-alive" , but haproxy will waits on the "tcp-request inspect-delay" which we set to 5s.

**Review by knobunc — CHANGES_REQUESTED** [review]

**Review by knobunc — APPROVED** [review]
/lgtm

**Comment by akostadinov** [comment]
while on it, is `tcp-request inspect-delay` adjustable or is it presently hardcoded?

**Comment by JacobTanenbaum** [comment]
@akostadinov tcp-request inspect-delay is hardcoded

**Comment by akostadinov** [comment]
Thank you. I think we need to make these parameters configurable. 5 seconds is a pretty big value. No need to wait for people with big installations to request that. But this is for another forum.

**Comment by bfallonf** [comment]
@JacobTanenbaum Thanks, this is good info to have in the docs.

I'm concerned people might not see it where it is, but I can't see a better location. Below is a section called Timeouts, that has almost literally no information. Would it be better to group the relevant timeout variable there? I don't mind making these changes in a followup PR if necessary.

**Comment by JacobTanenbaum** [comment]
@bfallonf are you recommending making a section about timeouts? I don't see one

**Comment by akostadinov** [comment]
@JacobTanenbaum , you can refer to the `Note` from the `ROUTER_SLOWLORIS_HTTP_KEEPALIVE` parameter description (such that users notice it). For me this would resolve the immediate issue. I don't mind if options are grouped better if needed though.

**Comment by bfallonf** [comment]
I spoke to @JacobTanenbaum over email that it'd be good to see what grouping together the timeout variables would look like. I'll merge this PR, then make a followup giving that a go. Thanks, all.

Also, I'll label this for 3.9 because that's the version the BZ above is set to. If I'm wrong, please correct me.

**Pull request merged**